### PR TITLE
New configure options

### PR DESCRIPTION
**Pull request by carloshenrq**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Added new ways to configure Hercules.
Now, can run configure with new parameteres, like:

`
./configure --enable-max-inventory=100 --enable-max-storage=800 --enable-max-zeny=1000000000 --enable-max-cart=200
`
Allowing change environment without using CPPFLAGS or chaning the source.

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** 

Master

[//]: # (Master? Slave?)

**Issues addressed:**

[//]: # (Issue Tracker Number if any.)

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2220)

File: configure.ac [hunk 1]
```diff
@@ -107,6 +107,389 @@ AC_ARG_ENABLE(
...
+	),
+	[enable_max_guardians="$enableval"],
+	[enable_max_guardians=""]
+)
+
+#
+# max_quest_objectives
+#
+AC_ARG_ENABLE(
+	[max_quest_objectives],
```
Comment (4144) [inline configure.ac]: this constants may break existing packets

File: configure.ac [hunk 2]
```diff
@@ -107,6 +107,389 @@ AC_ARG_ENABLE(
...
+	),
+	[enable_max_quest_objectives="$enableval"],
+	[enable_max_quest_objectives=""]
+)
+
+#
+# max_item_options
+#
+AC_ARG_ENABLE(
+	[max_item_options],
```
Comment (4144) [inline configure.ac]: this constants probably will break existing packets

**Comment by 4144** [comment]
this mean better not add option for this constants. if some one need, all constants can be changed by configure CPPFLAG option

**Comment by carloshenrq** [comment]
Ok, i'll review it and remove most of those.

**Comment by carloshenrq** [comment]
@4144 I've removed some config vars and combined them into one commit.

**Comment by dastgirp** [comment]
I kind of don't want more options in configure.
Adding #ifndef to mmo.h is fine. since we can redefine by following:
```
./configure CPPFLAGS="-DMAX_STORAGE=500 -DMAX_SKILL=1500"
```

**Comment by 4144** [comment]
now present useless merge commit, it should be removed too

but yes i think dont need too many options because already all possible to change by CPPFLAGS.

probably left only really usefull options?

**Comment by carloshenrq** [comment]
@dastgirp
I Only do this change, because i believe this way is more easly. This PR have selected to enter in next milestone, so...  


@4144
Ok, i'll remove the merge. About to many options...
The below vars, i don't mind if they're removed. I Just put in configure.ac because they are in #ifndef and i see no problem in put it...

	max_party
	max_bank_zeny
	max_fame
	max_cart
	max_skill_db
	max_skill_id
	max_skill_tree
	min_walk_speed
	max_walk_speed
	default_walk_speed
	max_guildposition
	max_guildexpulsion
	max_guildalliance
	max_guildlevel
	max_guildskill
	max_guardians



Other constants like, i think is more usefull then others above.

	sc_saving
	max_inventory
	max_chars
	max_zeny
	max_level
	base_guild_size
	max_guild
	
Brazilian servers, if not all, most of them change base_guild_size and max_guild...

I believe this way (using ./configure --enable-max-level=150), should less dificult for us and this should not 'disturb' other options.

**Comment by Asheraf** [comment]
@carloshenrq this type of configurations should be used only for major defines that is most likely in 50% of the cases people will change, (server type, client type, client version).. all other configurations can be done by editing directly in the file or using preprocessor definitions.

**Comment by carloshenrq** [comment]
@Asheraf The reason about this is: don't touch the source files, like we already do with plugins. One more time, i only made this change because i think it's more usefull and fast by this way... This shouldn't mess any other config.

**Comment by Asheraf** [comment]
As everyone have previously stated, adding these as configurations is not a great idea.. wrapping them in a `ifndef` is enough and that way preprocessor definitions can be used to set them.

**Comment by carloshenrq** [comment]
@Asheraf I agree with ifndef. Just for learning, there is a reason beyond "not great idea"?

**Comment by carloshenrq** [comment]
@4144 and/or @dastgirp what kind of changes this need? Only mmo.h?

**Comment by carloshenrq** [comment]
@4144 Can you review it again? Thank you =]

**Comment by Kenpachi2k13** [comment]
No activity for more than a year and it conflicts with current master. Close?

#clean_up_frenzy

**Comment by Kenpachi2k13** [comment]
*Resolved conflicts*

**Pull request merged**